### ISSUE TITLE
proton requires GLIBC_2.27 to run

### ISSUE THREAD

**Issue by toddyocum** [timeplus-io/proton#871]
**Describe what's wrong**

Unable to run proton-v1.6.3-Linux-x86_64 because of missing glibc requirement

**How to reproduce**

Try to run proton:

```
brawn:/base/data/files# ./proton-v1.6.3-Linux-x86_64 --version
./proton-v1.6.3-Linux-x86_64: /lib/libc.so.6: version `GLIBC_2.27' not found (required by ./proton-v1.6.3-Linux-x86_64)
```

Our embedded linux ( a customized older Fedora install ) libc version

```
brawn:/base/data/files# /lib/libc.so.6
GNU C Library (GNU libc) stable release version 2.24, by Roland McGrath et al.
```

**Error message and/or stacktrace**

**Additional context**

ClickHouse on the same box works, so was surprised that proton, which is built on top of/forked from ClickHouse, has a different run-time requirement. 

So might not be a bug, just a different requirement, but leaves us without a path forward at this time for trying out proton.

```
brawn:/base/data/files# clickhouse --version
ClickHouse local version 24.8.5.115 (official build).
```

**Comment by chenziliang** [comment]
Hi @toddyocum , thanks for reporting this issue. We will look into it. In the meanwhile, is docker container an option to unblock you first ?

**Comment by toddyocum** [comment]
Hi @chenziliang , docker isn't an option on our embedded linux (I wish it was)

But not a big deal, the PoC is a low priority atm. 

We are heavy users of ClickHouse and (lots of) MV's, and was curious to see what proton might offer over what we use.

**Comment by yokofly** [comment]
thanks, I can reproduce it with ubuntu1604 `docker run -it --rm ubuntu:16.04`
seems our glibc-compatible folder is not compatible in this case.
```
root@dcfd759831e4:~/timeplus/bin# /lib/x86_64-linux-gnu/libc.so.6
GNU C Library (Ubuntu GLIBC 2.23-0ubuntu11.3) stable release version 2.23, by Roland McGrath et al.
Copyright (C) 2016 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.
Compiled by GNU CC version 5.4.0 20160609.
Available extensions:
	crypt add-on version 2.1 by Michael Glad and others
	GNU Libidn by Simon Josefsson
	Native POSIX Threads Library by Ulrich Drepper et al
	BIND-8.2.3-T5B
libc ABIs: UNIQUE IFUNC
For bug reporting instructions, please see:
<https://bugs.launchpad.net/ubuntu/+source/glibc/+bugs>.
root@dcfd759831e4:~/timeplus/bin# 

```

**Comment by chenziliang** [comment]
Hi @toddyocum, thanks for the updates. Proton adds other functionalities like native JavaScript UDA and UDF, incremental join, incremental aggregation, incremental data deduplication, as well as streaming windows like hopping, sessionization, and high concurrent data ingest with small batches etc functions which are all complementary use cases to ClickHouse. We have quite a few users who deployed a Proton tier in front of ClickHouse doing ETL, streaming enchantments as well as light join, aggression and handle concurrent ingest before sending the data to ClickHouse.

For this glibc issue specifically, as the corresponding PR indicates, it is the embedded JavaScript V8 engine which depends on higher glibc version.


We can disable the JavaScript and provide a binary for you to experiment. Thanks!

**Comment by yokofly** [comment]
hi @toddyocum, thank you for reporting this issue. 
I have applied a patch for the V8 engine. The new binary is released here; you can take a trial: 
https://github.com/timeplus-io/proton/releases/v1.6.4

**Comment by tingfirst** [comment]
> Hi [@chenziliang](https://github.com/chenziliang) , docker isn't an option on our embedded linux (I wish it was)
> 
> But not a big deal, the PoC is a low priority atm.
> 
> We are heavy users of ClickHouse and (lots of) MV's, and was curious to see what proton might offer over what we use.

Hi @toddyocum ,  Again thanks for your interest in Proton! Compared to CH, Proton offers several exciting advantages—especially Incremental MV designed for high-EPS, low-latency workloads like network security monitoring.  Many of our users rely on it for  similar use cases. We’d love for you to join our Slack community (https://timeplus.com/slack) and share your thoughts! 

Ting / Timeplus
